random/s_final.go

```
@@ -0,0 +1,175 @@
+package main
+
+import (
+	"encoding/json"
+	"github.com/gin-gonic/gin"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"path"
+	"prac/bloomfilter"
+	"strconv"
+)
+
+type APIServer struct {
+	server *http.Server
+	id     []int
+	mb     map[int]bloomfilter.BloomFilter
+}
+
+type requestObject struct {
+	Data string `json:"data"`
+	Id   int    `json:"id"`
+}
+
+func main() {
+	s := &APIServer{server: &http.Server{}}
+	for i := 0; i < 10000; i++ {
+		s.id = append(s.id, i)
+	}
+	s.mb = make(map[int]bloomfilter.BloomFilter)
+	s.server.Addr = ":8002"
+	router := gin.Default()
+	s.server.Handler = router
+	router.GET("/new", s.handleNew)
+	router.POST("/put", s.handlePut)
+	router.POST("/mightcontain", s.handleMightContain)
+	router.GET("/delete/:id", s.handleDelete)
+	log.Println("[INFO] running apiserver on port:", 8002)
+	err := s.server.ListenAndServe()
+	if err != nil {
+		panic(err)
+	}
+}
+
+func (s *APIServer) handleNew(c *gin.Context) {
+	if len(s.id) == 0 {
+		c.JSON(http.StatusBadRequest, gin.H{
+			"result":  "Failure",
+			"message": "Cannot create a new Bloomfilter : Please delete an existing Bloomfilter",
+		})
+
+	}
+	idNumber := s.id[0]
+	s.id = s.id[1:]
+	s.mb[idNumber], _ = bloomfilter.NewBloomFilter(1000)
```

Do not ignore the error here